### ISSUE TITLE
How to classify bottle

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#712]
**Is your feature request related to a problem? Please describe.**
- Based on https://github.com/mars-sim/mars-sim/issues/710, it makes sense to create bottle as a container that would hold water

**Describe the solution you desire to see**
- Currently, it seems fitting for creating Bottle under GenericContainer (which extends Equipment) 

**Describe alternatives you've considered**
- Treat water bottle as a specialized Item and with an assumed fixed volume of water inside. 
- The water inside will need to be tracked 
- Alternatively, the whole bottle (and the water content inside) may be consumed only once

**Specifications  (please complete the following information):**
 - pre-3.4.0 build 7455

**Additional context**
- Each person may be assigned with the same bottle for holding water or liquid dessert such as soymilk, cranberry juice, etc.

**Comment by mokun** [comment]
@bevans2000 

Currently, the containers we have, namely, bags, gas canister, specimen box, etc. is different from EVA suits.

```
/**
 * The EquipmentType enum class is used for distinguishing between various type of equipments
 */
public enum EquipmentType {
	
	// non-container
	EVA_SUIT			(Msg.getString("EquipmentType.EVASuit")), //$NON-NLS-1$ 

	// Container 
	BAG 				(Msg.getString("EquipmentType.bag")), //$NON-NLS-1$
	BARREL 				(Msg.getString("EquipmentType.barrel")), //$NON-NLS-1$
	GAS_CANISTER		(Msg.getString("EquipmentType.gasCanister")), //$NON-NLS-1$
	LARGE_BAG			(Msg.getString("EquipmentType.largeBag")), //$NON-NLS-1$
	SPECIMEN_BOX		(Msg.getString("EquipmentType.specimenBox")); //$NON-NLS-1$
	
```

It still makes sense to call EVASuit as a Equipment object, however, containers almost don't need to be tracked individually.

Now that we are to create `bottle`, is it better off we create a new enum class called ContainerType to denote them ?

**Comment by bevans2000** [comment]
I agree with your thoughts. Containers are not modelled correctly.

Containers are over modelled being sub-classed from ```Unit``` as they are passive not active objects. In fact they have more in common with ```Part```, e.g. hammer is a Part. However the problem is we don't track individual Part items; they are too numerous. But Containers need to the tracked as they are ```ResourceHolders```. 

My thought is there is a need for an ```Equipment``` which represents items that need to be individually tracked but it should NOT be a ```Unit```. Being a Unit increases the size of the object and adds behaviors that an Equipment does not need. There are more ```Equipment``` instances that anything else so the impact on the overall memory consumption is high. This would also drop the Equipment map from the ```UnitManager``` which is a good thing. ```Equipment``` would become a much smaller object which is good.

Dropping the Unit behavior does come with some drawbacks:

- An Equipment must always be attached to a ```EquipmentOwner```. We can't drop them on the surface and leave them.
- They would not have an individual ```Coordinate``` but if they are always attached does it matter?
- No direct associated Settlements but as above that doesn't matter.

Nevertheless, I imagine changing could throw up some problems. So it needs to be done on a branch but it would be a good thing to do.

**Comment by mokun** [comment]
> My thought is there is a need for an `Equipment` which represents items that need to be individually tracked

Yes. There's a need for having `Equipment` class. EVASuit is a wonderful example. So is `Robot`. 

We just haven't thought of how we best use Equipment class to represent each as an object. 

And we haven't even modeled in-depth how a robot needs maintenance and would also malfunction and repair, etc.

May be one of the next thing I want to do is to give a better modeling fidelity to how each EVA suit operates and individualize the behavior of each EVASuit object.
 
So in our case, the fact that currently each container, such as a large bag, a gas canister, or specimen box are Equipment are overkill, since we don't care much about how one specimen box differs from another.

Can we have one and only one SpecimenBox object that has a map or a list of boxes which will track how much stuff in each box as follows:  

```
class SpecimenBox extends GenericContainer {
     
      Map<Integer, Box> boxes = new HashMap<>();

      class SpecimenBox() {
      ...
      }

      private class Box(Resource resource, double quantity) {
      ...
      }
}
```

Currently, each specimen box is a GenericContainer object.

**Comment by bevans2000** [comment]
Creating a super Container that contains individual boxes I see as being more complex. One benefit of the current approach is it easy to understand and flexible. Having a separate manager will just add more complexity and specialisation. Now the ```EquipmentOwner``` is in place we can remove the Unit subclass without making a massive impact.
Also ```Robot``` should not be an Equipment but directly subclassed from Unit. We do treat Robots different.
